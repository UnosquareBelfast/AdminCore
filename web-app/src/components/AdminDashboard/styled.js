import styled from 'styled-components';

export const InnerLayout = styled.div`
  width: 100%;
  h2 {
    margin: 0;
  }

  .columnWrap {
    display: flex;
    margin-top: 20px;
    border-radius: 8px;

    div {
      display: flex;
      flex-direction: column;
      align-items: center;
      padding: 20px 10px;
      width: 100%;
      border: 1px solid ${({ theme }) => theme.colours.grey};
      border-right: none;

      :first-of-type {
        border-radius: 8px 0 0 8px;
      }
      :last-of-type {
        border-radius: 0 8px 8px 0;
        border-right: 1px solid ${({ theme }) => theme.colours.grey};
      }

      h3 {
        margin: 0 0 10px 0;
      }

      a {
        color: ${({ theme }) => theme.colours.unoBlue};
        font-weight: ${props => props.theme.fonts.weight.bold};
        text-decoration: none;
        margin-bottom: 5px;

        :hover {
          color: ${({ theme }) => theme.colours.darkBlue};
          text-decoration: underline;
        }
      }
    }
  }
`;
