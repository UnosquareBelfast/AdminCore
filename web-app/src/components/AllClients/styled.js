import styled from 'styled-components';

export const CTA = styled.div`
  display: flex;
  flex-direction: row-reverse;
  padding: 20px 0;

  button {
    width: auto;
    margin-left: 10px;
  }
`;

export const ActiveDot = styled.div`
  display: inline-block;
  position: relative;
  top: -2.5px;
  margin-right: 7px;
  height: 7px;
  width: 7px;
  border-radius: 7px;
  background: ${({ theme, active }) =>
    active ? theme.colours.green : theme.colours.red};
`;
