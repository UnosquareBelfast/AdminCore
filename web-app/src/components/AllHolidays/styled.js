import styled from 'styled-components';

export const HolidayTable = styled.table`
  width: 100%;

  tr:nth-child(even) {
    background-color: ${props => props.theme.colours.lightgrey};
  }

  tr td:last-of-type {
    text-align: right;
  }

  th {
    text-align: left;
  }

  button {
    border: none;
    background: ${props => props.theme.colours.unoBlue};
    color: white;
    padding: 3px 10px;
    margin: 2px 5px 2px 0;
    border-radius: 3px;
    cursor: pointer;
  }

  button:hover {
    background: ${props => props.theme.colours.darkBlue};
  }
`;

export const StatusDot = styled.div`
  display: inline-block;
  position: relative;
  top: -2.5px;
  margin-right: 7px;
  height: 7px;
  width: 7px;
  border-radius: 7px;
  background: ${({ theme, status }) => theme.holidayStatus[status]};
`;
