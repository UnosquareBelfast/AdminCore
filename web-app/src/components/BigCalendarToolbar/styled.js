import styled from 'styled-components';

export const StyleContainer = styled.div`
  display: flex;
  flex-direction: column;
  margin: 0 0 15px 0;
  @media (min-width: ${props => props.theme.mediaQueries.md}) {
    flex-direction: row;
    justify-content: space-between;
  }

  h2 {
    margin: 0 0 0px 0;
  }
  p {
    margin: 8px 0 0 0;
    text-transform: uppercase;
    color: ${({ theme }) => theme.colours.darkGrey};
  }

  button {
    cursor: pointer;
    outline: none;
    height: 40px;
    padding: 10px;
    margin-top: 10px;
    background-color: ${({ theme }) => theme.colours.lightgrey};
    border: 1px solid ${({ theme }) => theme.colours.grey};
    color: ${({ theme }) => theme.colours.darkGrey};
    border-right: 0;
    :first-of-type {
      border-radius: 4px 0 0 4px;
    }
    :last-of-type {
      border-radius: 0 4px 4px 0;
      border-right: 1px solid ${({ theme }) => theme.colours.grey};
    }
    :hover {
      background-color: ${({ theme }) => theme.colours.unoBlue};
      border-color: ${({ theme }) => theme.colours.unoBlue};
      color: white;
    }
    :active {
      background-color: ${({ theme }) => theme.colours.darkBlue};
      border-color: ${({ theme }) => theme.colours.darkBlue};
    }
  }
`;
