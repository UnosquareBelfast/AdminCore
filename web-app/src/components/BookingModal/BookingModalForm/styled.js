import styled from 'styled-components';

export const NoticeAlert = styled.div`
  margin-top: 20px;
  margin-bottom: 10px;
  span {
    font-weight: bold;
  }
  svg {
    position: relative;
    top: 10px;
    margin-right: 8px;
  }
  p {
    margin: 0 0 5px 0;
    :last-of-type {
      margin-left: 26px;
    }
  }
`;
