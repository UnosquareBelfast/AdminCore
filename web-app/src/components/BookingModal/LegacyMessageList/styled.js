import styled from 'styled-components';

export const Ul = styled.ul`

    --border-style: 2px solid #d6d6d6;
    --employee-1-message-color: #0eb5d1;
    --employee-2-message-color: #0ecaea;

    li {
        list-style-type: none;
        margin: 10px;
        border-radius: 25px;
        width: 75%;
        padding-left: 30px;
        padding-top: 1px;
        padding-bottom: 1px;
        word-wrap:break-word;
    }

    h1 {
        margin: 10px;
        position: relative;
        color:white;
    }

    h3 {
        text-transform: none;
        margin: 7px 10px 13px 7px;
        color: white;
        font-family: Open Sans,sans-serif;
        padding-right: 20px;
        padding-left: 20px;
    }
    
    height: 300px; 
    overflow: auto;
    border: var(--border-style);
    text-align: center;
    
    .employee-message-2{
        background-color: var(--employee-2-message-color);
    }
    .employee-message-1{
        background-color: var(--employee-1-message-color);
    }

    .legacy-message-container {
        display: flex;
        justify-content: space-between;
    }

`;
