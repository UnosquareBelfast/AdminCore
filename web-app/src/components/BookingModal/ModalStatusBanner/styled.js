import styled from 'styled-components';

export const Banner = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-between;
  color: white;
  background: ${({ theme, status }) => theme.holidayStatus[status]};
  padding: 10px;
  margin-bottom: 20px;
  border-radius: 3px;
  p {
    margin: 4px 0 0 0;
  }
  svg {
    margin-right: 5px;
  }

  .cancelEvent {
    cursor: pointer;
    border: 1.5px solid white;
    padding: 10px; 15px;
    border-radius: 3px;
  }
`;
