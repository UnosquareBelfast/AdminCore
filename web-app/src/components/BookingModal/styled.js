import styled from 'styled-components';

export const StyleContainer = styled.div`
  margin: 20px;
  @media (min-width: ${props => props.theme.mediaQueries.lg}) {
    margin: 40px;
  }
  #closeBookingModal {
    position: absolute;
    cursor: pointer;
    top: 5px;
    font-weight: bold;
  }
  h1 {
    margin: 0 0 10px 0;
  }

  h4 {
    margin: 0;
  }
`;

export const FormContainer = styled.div`
  position: relative;

  form {
    @media (min-width: ${props => props.theme.mediaQueries.lg}) {
      margin-right: -10px;
      margin-left: -10px;
    }

    .date {
      transition: all 300ms;
      visibility: visible;
      @media (min-width: ${props => props.theme.mediaQueries.lg}) {
        margin: 0 0 20px 0;
        display: inline-block;
        width: 50%;
        padding: 0 10px;
        box-sizing: border-box;
        vertical-align: bottom;
      }
    }

    .select {
      @media (min-width: ${props => props.theme.mediaQueries.lg}) {
        margin: 20px 10px;
        width: calc(100% - 20px);
      }
    }

    .checkbox {
      @media (min-width: ${props => props.theme.mediaQueries.lg}) {
        margin: 0 10px 20px 10px;
      }
    }

    > div:last-child {
      @media (min-width: ${props => props.theme.mediaQueries.lg}) {
        width: calc(50% - 10px);
        box-sizing: border-box;
        margin-left: calc(50% + 10px);
      }
    }
  }
`;
