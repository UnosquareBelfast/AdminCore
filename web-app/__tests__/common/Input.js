import React from 'react';
import { shallow, mount } from 'enzyme';
import { Input } from '../../src/components/common';
import { theme } from '../../src/styled';

const getInputEl = () => {
  theme.colours.grey = '#d6d6d6';
  return (
    <Input
      theme={theme}
      type="input"
      htmlAttrs={{
        type: 'text',
        name: 'test',
        placeholder: 'Test placeholder value',
      }}
      value="Hello"
      label="Test input field:"
    />
  );
};

xdescribe('Input Component', () => {
  it('renders correctly', () => {
    const wrapper = shallow(getInputEl());
    expect(wrapper.exists()).toBe(true);
    expect(wrapper).toMatchSnapshot();
  });

  it('Test onchange of a text input', done => {
    const wrapper = mount(getInputEl());
    const input = wrapper.find('input[name="test"]');

    input.simulate('focus');
    input.simulate('change', { target: { value: 'Changed' } });
    expect(input.get(0).value).to.equal('Changed');

    done();
  });
});
