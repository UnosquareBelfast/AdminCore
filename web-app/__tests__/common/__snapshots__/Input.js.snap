// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Component renders correctly 1`] = `
<Input
  changed={[Function]}
  focus={false}
  htmlAttrs={
    Object {
      "name": "test",
      "placeholder": "Test placeholder value",
      "type": "text",
    }
  }
  label="Test input field:"
  rules={null}
  theme={
    Object {
      "colours": Object {
        "darkBlue": "#0a98af",
        "darkGrey": "#3a3939",
        "darkRed": "#c4412b",
        "green": "#35c375",
        "grey": "#d6d6d6",
        "lightBlue": "#daffff",
        "lightRed": "#fda49a",
        "lightgrey": "#f7f7f7",
        "red": "#ff3434",
        "unoBlue": "#0eb5d1",
        "white": "#ffffff",
        "yellow": "#ff9b34",
      },
      "fonts": Object {
        "fallback": "sans-serif",
        "headers": "Oswald",
        "main": "Open Sans",
      },
      "holidayStatus": Object {
        "1": "#ff9b34",
        "2": "#35c375",
        "3": "#ff3434",
        "4": "#0eb5d1",
        "5": "#3469ff",
        "6": "#551A8B",
        "7": "#333333",
      },
    }
  }
  touched={false}
  type="input"
  valid={false}
  validateForm={[Function]}
  value="Hello"
/>
`;
